ipatests/test_integration/test_sssd.py

```diff
@@ -113,3 +117,60 @@ def test_extdom_group(self):
             client_conf_backup.restore()
             tasks.clear_sssd_cache(self.master)
             tasks.clear_sssd_cache(client)
+
+    def test_external_group_paging(self):
+        """SSSD should fetch external groups without any limit.
+
+        Regression test for https://pagure.io/SSSD/sssd/issue/4058
+        1: Add external groups more than limit.
+        2: Run the command id aduser@ADDOMAIN.COM
+        3: sssd should retrieve all the external groups.
+        """
+        cmd = self.master.run_command(['sssd', '--version'])
+        sssd_version = platform_tasks.parse_ipa_version(
+            cmd.stdout_text.strip())
+        if LooseVersion(sssd_version.version) <= LooseVersion('1.16.3'):
```

Comment:
Do you really need LooseVersion or would it work with:
```
if sssd_version <= platform_tasks.parse_ipa_version('1.16.3'):
```

Reply:
sssd_version <= platform_tasks.parse_ipa_version('1.16.3'): this will compare string with a string.
That's why it needs LooseVersion, as sssd_version.version with type string.

Reply:
Hi @amore17 

> sssd_version <= platform_tasks.parse_ipa_version('1.16.3'): this will compare string with a string.

I'm afraid I disagree here. `parse_ipa_version` is using `pkg_resources.parse_version()` and this [method](https://setuptools.readthedocs.io/en/latest/pkg_resources.html#parsing-utilities) returns an object that can be compared/sorted.
Using `sssd_version.version` means using the string representation and the benefit of the parsing is lost.